### DISCUSSION TITLE
Refresh page on width change

### DISCUSSION THREAD

**Discussion by OzGav** [ExperienceLovelace/ha-floorplan#472] [Ideas]
As discussed here https://github.com/ExperienceLovelace/ha-floorplan/discussions/320#discussioncomment-11080799 it would be great if the floorplan was refreshed automatically on device orientation or browser width change.

**Comment by exetico** [comment]
Hi OzGav

Thank you for submitting this. I've made some initial thought about this.

Going with window.innerWidth as a trigger point, will have some caveats. 

Here's my floorplan, but Home Assistant did hide the side-menu, due to the window size:
![image](https://github.com/user-attachments/assets/4d1cff67-3acc-485f-a6e2-270fbbda8905)

But here's it's not hidden:
![image](https://github.com/user-attachments/assets/ab7fa9a7-2361-4e36-b840-7f9427e601bf)

But the svg itself does have another size. So using window.innerWidth will generate a few funny situations.

Also; I personally debug with my Developer Tool in the left or right side of the window, and not in a pop-up. Triggering a re-draw in that situation, will make it a big tricky to debug, if the user are not well-known in the Developer Tool, and knows that "popping out the Dev Tool" will solve it. It will also add a bit more complexity, as we'd need to decide how often we'll allow the reload of the floorplan solution to be triggered, on basis of how aggressive the window changes was/is.

Maybe it's a good idea to stick to a eventListener of the device-orientation, as it's a perfect sweet-spot, and won't require much on the table to be discussed/targeted in.

Do you think a orientation-triggered "redraw" of a floorplan will have a positive impact?

**Reply by OzGav** [reply]
Yes I do. Certainly that is my use case. I think go with that and see if anyone else has bigger ideas…?
